docs/DeveloperGuide.md

```diff
@@ -275,6 +275,106 @@ as values.<br>
+
+`ContactedInfo` objects have the following characteristics:
+
+* A `ContactedInfo` object has two private data members (final), `recentDate` of `RecentDate` object representing that date of interaction,
+and `description` of `Description` object, representing the description of the interaction. Both objects would distinguish one ContactedInfo object from another.
+* Any two `ContactedInfo` objects are not unique if both `description` and `recentDate` is equal.
+* `ContactedInfo` objects can be sorted, and the sorted order is the reverse ordering of the `RecentDate`.
+
+###### RecentDate
+
+`RecentDate` is an object that stores information regarding the interaction date for `ContactedInfo`. `RecentDate` object inherits from `DocumentedDate` object.
+The sequence diagram below shows the creation of a RecentDate object.
+
+<img src="images/RecentDateCreationSequenceDiagram.png" width="650" />
+
+`RecentDate` objects have the following characteristics:
+
+* A `RecentDate` object has two private data members (final), `date` representing the date of interaction as a `LocalDate` object,
+and `value` of `String` format representing the date in `YYYY-MM-DD` form.
+* input to create a `recentDate`object needs to be the correct format (`YYYY-MM-DD`).
+* Any two `recentDate` objects are not unique if both `recentDate` represents the same date.
+* `recentDate` objects can be sorted, and the sorted order is the reverse ordering of their `LocalDate`.
+
+`RecentDate` implements the following operations.
+
+* `parse(String parsedDate)`<br>
+  Creates a new `RecentDate` using a String. String has to have the format `YYYY-MM-DD`.
+
+* `defaultRecentDate()`<br>
+  Returns today's date as a `RecentDate` object.
+
+###### Description
+
+`Description` object represents the description of the recent interaction. Description gets invoked in the ParserUtil parseContactedInfo method.
+The sequence diagram below shows what happens when a Description object is instantiated.
+
+<img src="images/DescriptionSeqDiagram.png" width="450" />
+
+`Description` objects have the following characteristics:
+
+* `Description` needs to be alphanumeric (only letters and numerals are allowed), and should not exceed 280 characters.
+* Contains one public data member (final) `value` of `String` object, representing the description of the `Description` object,
+which can be used to distinguish itself from other `Description` object.
+
+`Description` implements the following operations.
+
+* `parse(String parsedDate)`<br>
+  Creates a new `RecentDate` using a String. String has to have the format `YYYY-MM-DD`.
+
+* `defaultDesc()`<br>
+  Returns a `Description` object containing the String with the description being "First Interaction".
+
+* `isValidDescription(String test)` <br>
+  Checks if the parsed String is a valid input. Returns true if the input String is alphanumeric and does not exceed 280 characters,
+  otherwise false.
+
+#### Design Consideration
```

Comment:
Design considerations is to justify why you use a certain implementation. From what you write I can't really tell why you're using one implementation over the other

Reply:
Can consider elaborating more in the cons or using more specific pros rather than just "Easy to handle"